.devops/deploy-pipelines.yml

```diff
@@ -0,0 +1,203 @@
+              #  ex: Build.SourceBranch=refs/heads/$(DEFAULT_BRANCH) --> $(DEFAULT_BRANCH)
+              #  ex: Build.SourceBranch=refs/tags/v1.2.3-RELEASE --> v1.2.3-RELEASE
+              gitReference: ${{ replace(replace(variables['Build.SourceBranch'], 'refs/tags/', ''), 'refs/heads/', '') }}
+          - script: |
+              yarn predeploy
+            displayName: 'Build'
+  
+          # Copy application to  
+          - task: CopyFiles@2
+            inputs:
+              SourceFolder: '$(System.DefaultWorkingDirectory)'
+              TargetFolder: '$(System.DefaultWorkingDirectory)/bundle'
+              Contents: |
+                **/*
+                !.git/**/*
+                !.vscode/**/*
+                !.devops/**/*
+                !.conf-testenv/**/*
+                !.scripts/**/*
+                ![.]*
+                !**/*.js.map
+                !**/*.ts
+                !*.md
+                !docker-compose.yml
+                !DockerFile
+                !Dangerfile.ts
+                !jest.config.js
+                !local.settings.json
+                !test
+                !tslint.json
+                !tsconfig.json                
+            displayName: 'Copy deploy files'
+          
+          - publish: $(System.DefaultWorkingDirectory)/bundle
+            artifact: Bundle
+
+  # Deploy 
+  - stage: 'Deploy_to_${{ parameters.environment }}'
+    dependsOn:
+      - Prepare_artifact
+    jobs:
+      - job: 'do_deploy_staging_slot_and_swap'
+        condition: eq(variables.DEPLOY_TYPE, 'staging_slot_and_swap')
+        steps:
+          - checkout: none
+          - download: current
+            artifact: Bundle
+
+          - task: AzureWebApp@1
+            inputs:
+              azureSubscription: '$(AZURE_SUBSCRIPTION)'
+              resourceGroupName: '$(WEB_APP_RESOURCE_GROUP_NAME)'
+              appType: 'webApp'
+              appName: '$(WEB_APP_NAME)'
+              package: '$(Pipeline.Workspace)/Bundle'
+              deploymentMethod: 'auto'
+              deployToSlotOrASE: true
+              slotName: 'staging'
+            displayName: 'Deploy to staging slot'
+
+          - template: templates/rest-healthcheck/template.yaml@pagopaCommons
+            parameters:
+              azureSubscription: '$(AZURE_SUBSCRIPTION)'
+              appName: '$(WEB_APP_NAME)'
+              endpoint: '$(HEALTHCHECK_ENDPOINT)'
+              endpointType: 'private'
+              containerInstanceResourceGroup: '$(HEALTHCHECK_CONTAINER_RESOURCE_GROUP_NAME)'
+              containerInstanceVNet: '$(HEALTHCHECK_CONTAINER_VNET)'
+              containerInstanceSubnet: '$(HEALTHCHECK_CONTAINER_SUBNET)'
+
+          - task: AzureAppServiceManage@0
+            inputs:
+              azureSubscription: '$(AZURE_SUBSCRIPTION)'
+              resourceGroupName: '$(WEB_APP_RESOURCE_GROUP_NAME)'
+              webAppName: '$(WEB_APP_NAME)'
```

Comment:
AFAIK we don't have staging and production slot by using app service. Right? @pasqualedevita

Reply:
this new pipeline handle both cases setting these vars in gitops:
UAT_DEPLOY_TYPE
PROD_DEPLOY_TYPE

values:
1. `production_slot`
2. `staging_slot_and_swap`

My opinion, I prefer to have all pipeline with same structure